### ISSUE TITLE
Error while running docker_build

### ISSUE THREAD

**Issue by darthgera123** [adobe/sbmc#13]
Hi I followed all instructions to run this repo but im getting the following error when im trying to run ```docker_build```. Please help
``` /sbmc_app/halide/include/HalidePyTorchHelpers.h:58:36: error: expected constructor, destructor, or type conversion before ‘(’ token
 AT_FORALL_SCALAR_TYPES_WITH_COMPLEX(HL_PT_DEFINE_TYPECHECK);
                                    ^
/sbmc_app/halide/include/HalidePyTorchHelpers.h: In instantiation of ‘Halide::Runtime::Buffer<scalar_t> Halide::PyTorch::wrap(at::Tensor&) [with scalar_t = float]’:
build/temp.linux-x86_64-3.7/scatter2gather_cpu_float32.pytorch.h:15:74:   required from here
/sbmc_app/halide/include/HalidePyTorchHelpers.h:67:46: warning: ‘T* at::Tensor::data() const [with T = float]’ is deprecated [-Wdeprecated-declarations]
     scalar_t* pData  = tensor.data<scalar_t>();
                                              ^
In file included from /sbmc_app/anaconda/lib/python3.7/site-packages/torch/include/ATen/Tensor.h:11:0,
                 from /sbmc_app/anaconda/lib/python3.7/site-packages/torch/include/ATen/Context.h:4,
                 from /sbmc_app/anaconda/lib/python3.7/site-packages/torch/include/ATen/ATen.h:5,
                 from /sbmc_app/anaconda/lib/python3.7/site-packages/torch/include/torch/csrc/api/include/torch/types.h:3,
                 from /sbmc_app/anaconda/lib/python3.7/site-packages/torch/include/torch/csrc/api/include/torch/data/dataloader_options.h:4,
                 from /sbmc_app/anaconda/lib/python3.7/site-packages/torch/include/torch/csrc/api/include/torch/data/dataloader/base.h:3,
                 from /sbmc_app/anaconda/lib/python3.7/site-packages/torch/include/torch/csrc/api/include/torch/data/dataloader/stateful.h:3,
                 from /sbmc_app/anaconda/lib/python3.7/site-packages/torch/include/torch/csrc/api/include/torch/data/dataloader.h:3,
                 from /sbmc_app/anaconda/lib/python3.7/site-packages/torch/include/torch/csrc/api/include/torch/data.h:3,
                 from /sbmc_app/anaconda/lib/python3.7/site-packages/torch/include/torch/csrc/api/include/torch/all.h:4,
                 from /sbmc_app/anaconda/lib/python3.7/site-packages/torch/include/torch/extension.h:4,
                 from build/temp.linux-x86_64-3.7/wrapper.cpp:1:
/sbmc_app/anaconda/lib/python3.7/site-packages/torch/include/ATen/core/TensorBody.h:303:7: note: declared here
   T * data() const {
       ^
error: command 'gcc' failed with exit status 1
```

**Comment by darthgera123** [comment]
@muhqu could you please help

**Comment by muhqu** [comment]
@darthgera123 sorry, I'm not involved in this project. You might want to reach out to actual [contributors](https://github.com/adobe/sbmc/graphs/contributors).

**Comment by darthgera123** [comment]
@muhqu  thanks for replying and sorry for bothering. @mgharbi  could you please look into it once

**Comment by darthgera123** [comment]
anyone?

**Comment by mgharbi** [comment]
This is due to a change of API in Pytorch 1.3, and the Halide bindings are not up to date, yet. I will look into this

**Comment by darthgera123** [comment]
thanks. meanwhile what version of pytorch did you use, ill try to compile it with that

**Comment by mgharbi** [comment]
The breaking change happens in >= 1.3. So, 1.2 or earlier should work.

**Comment by mgharbi** [comment]
I've changed the torch requirements in the `dockerfile` and `setup.py`. This should be fixed in 6a38abbe9df409ea4d44398dc610d1f3e8072392
